executor/index_merge_reader.go

```diff
@@ -642,11 +669,16 @@ func (w *partialTableWorker) extractTaskHandles(ctx context.Context, chk *chunk.
 		memUsage += memDelta
 		w.memTracker.Consume(memDelta)
 		for i := 0; i < chk.NumRows(); i++ {
-			handle, err := handleCols.BuildHandle(chk.GetRow(i))
+			r := chk.GetRow(i)
+			handle, err := handleCols.BuildHandle(r)
 			if err != nil {
 				return nil, nil, err
 			}
 			handles = append(handles, handle)
+			// reduce memory usage when len(w.byItems) == 0
+			if len(w.byItems) != 0 {
+				retChk.AppendRow(r)
```

Comment:
Why bother AppendRow again? Because `CopyConstruct()` will copy all data of `chk`